subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java

```diff
@@ -127,8 +128,11 @@ public void compile(List<String> files, CompilationUnit cu) {
                         } else {
                             // When annotation processing isn't required, it's better to add the Groovy stubs as part of the source path.
                             // This allows compilations to complete faster, because only the Groovy stubs that are needed by the java source are compiled.
-                            spec.getCompileOptions().getCompilerArgs().add("-sourcepath");
-                            spec.getCompileOptions().getCompilerArgs().add(stubDir.getAbsolutePath());
+                            FileCollection sourcepath = new SimpleFileCollection(stubDir);
+                            if (spec.getCompileOptions().getSourcepath() != null) {
+                                sourcepath = spec.getCompileOptions().getSourcepath().plus(sourcepath.getAsFileTree());
```

Comment:
Oops, this is obviously completely wrong. Looks like this is completely untested code (at least not by the checks done in the Travis-CI build)